lib/WorseReflection/Core/Inference/Resolver/MemberAccessExpressionResolver.php

```diff
@@ -104,7 +103,7 @@ public static function infoFromMemberAccess(NodeContextResolver $resolver, Frame
                     );
                 }
 
-                $types[] = $declaringClass;
+                $types[] = $subType;
```

Comment:
this change isa bit risky but I think it's the correct behavior. previously the "container type" would be resolved to the declaring type (so for example the "trait" that declared it). this caused the bug because the Trait reflection does not include virtual methods (they are grafted onto the instance class).

what this means is that if you reflect the offset for the member call, the "container type" will be the parent class, trait or interface, which might not be what you'd expect.